src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -1980,7 +1980,29 @@ private void LexSyntaxTrivia(bool afterFirstToken, bool isTrailing, ref SyntaxLi
                     case '#':
                         if (_allowPreprocessorDirectives)
                         {
-                            this.LexDirectiveAndExcludedTrivia(afterFirstToken, isTrailing || !onlyWhitespaceOnLine, ref triviaList);
+                            if (isTrailing || !onlyWhitespaceOnLine)
+                            {
+                                // Directives cannot be in trailing trivia - parse as skipped tokens trivia instead.
+
+                                var savePosition = TextWindow.Position;
+
+                                this.LexDirectiveAndExcludedTrivia(afterFirstToken, afterNonWhitespaceOnLine: true, ref triviaList);
+
+                                var text = TextWindow.Text.GetSubText(TextSpan.FromBounds(savePosition, TextWindow.Position));
+
+                                var error = new SyntaxDiagnosticInfo(
+                                    offset: 0,
+                                    width: Math.Min(1, text.Length),
```

Comment:
long widths like this are undesirable.  what would be better would just be a diagnostic of length 1 under the `#`.

Reply:
That's what it is; the min just ensures it goes to 0 if the text is empty (not sure if that can actually happen)

Reply:
1.  Oh duh, i misread min/max
2. You clearly can't have 0 length text, since you literally saw the # character :).  You can just use '1' here.

Do make sure that the error shows up on the right char though if there are things like spaces before the `#`.